src/main/java/hudson/plugins/claim/ClaimEmailer.java

```diff
@@ -56,7 +56,7 @@ public static void sendInitialBuildClaimEmailIfConfigured(@Nonnull User claimedB
             throws MessagingException, IOException {
 
         ClaimConfig config = ClaimConfig.get();
-        if (config.getSendEmails() && MAILER_LOADED) {
+        if (config.getSendEmails() && MAILER_LOADED && assignee.getFullName() != assignedBy) {
```

Comment:
`*Message` classes (below) have a mustSend method indicating if the mail is to be sent.
The code should be integrated to this method.
Also, the code is already trying to prevent this situation. I would be interested to know what user setup was creating the issue you are trying to solve.